src/Storefront/Theme/Validator/SCSSValidator.php

```diff
@@ -212,4 +203,86 @@ private static function extractSCSSvars(string $value): array
 
         return $vars;
     }
+
+    private static function isHex(string $hexCode): bool
+    {
+        preg_match('/^[a-f0-9]*$/i', $hexCode, $parsed);
+
+        return isset($parsed[0]) && $parsed[0] === $hexCode;
+    }
+
+    /**
+     * Validates if the given HSL code is valid.
+     *
+     * It supports both classic and modern syntax.
+     * Will forward to isHSLA for classic syntax with 'hsla'.
+     */
+    private static function isHSL(string $hslCode): bool
+    {
+        if (!str_starts_with($hslCode, 'hsl')) {
+            return false;
+        }
+
+        if (str_starts_with($hslCode, 'hsla')) {
+            return self::isHSLA($hslCode);
+        }
+
+        $hue = '(?:360(?:\.0+)?|3[0-5]\d(?:\.\d+)?|[12]\d{2}(?:\.\d+)?|[1-9]?\d(?:\.\d+)?)';
```

Comment:
where did you find the regexes? sometimes we link to examples on https://regex101.com
or if this is something official, you can link the source?

Reply:
oh that is nice. can't we use that and fallback checking the modern syntax? and best would be a contribution to Symfony to add the missing things :innocent:

Reply:
Adding another dependency that doesn't even do what we want for 4 little helper functions that already work as expected? 😑

Reply:
:man_shrugging: do not invent the wheel again :grin: 

I approved, do as you like. I am fine with both